float64 x
float64 y
float64 x_vel
float64 z_vel
